### PR TITLE
[121] Implement features service

### PR DESCRIPTION
**Pull request by darokel**
### Context

- https://trello.com/c/E47aN3UJ/121-implement-features-service

### Changes proposed in this pull request

- Implement a service module to handle feature flag checking
- Add basic feature flag smoke test

### Guidance to review

- Toggle the sample feature flag on/off locally and assert the text in the home page shows/hides

### CODE REVIEW

**Comment by darokel** [comment]
[Implement features service](https://trello.com/c/E47aN3UJ/121-implement-features-service)

File: config/settings.yml [hunk 1]
```diff
@@ -1 +1,3 @@
-basic_auth_required: true
\ No newline at end of file
+basic_auth_required: true
+features:
+  home_text: false
```
Comment (misaka) [inline config/settings.yml:3]: newline pluuzzzz!

Comment (aliuk2012) [inline config/settings.yml:3]: could we add an inline comment to say what the purpose is of this config setting

Comment (darokel) [reply config/settings.yml:5]: done https://github.com/DFE-Digital/register-trainee-teacher-data/pull/44/commits/88429b4afbb13f436fdf818a6b1a2029536d9957

File: spec/features/view_pages_spec.rb [hunk 2]
```diff
@@ -1,13 +1,13 @@
 require "rails_helper"
 
-RSpec.feature "view pages", type: :system do
+RSpec.feature "view pages", type: :feature do
```
Comment (aliuk2012) [inline spec/features/view_pages_spec.rb:3]: Do we need to specify the type for every feature spec in `spec/features/`. I added a support method for components to automatically add the type to specs found within `app/components`

Comment (darokel) [reply spec/features/view_pages_spec.rb:3]: No you don't, this is already baked into rspec and will be set automatically if Capybara is required so I don't think we need to do/add anything here.

**Review by aliuk2012 — APPROVED** [review]

File: app/services/feature_service.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+module FeatureService
+  class << self
+    def enabled?(feature_name)
+      return false if Settings.features.blank?
+
+      # Supports nested feature segments
```
Comment (gpeng) [inline app/services/feature_service.rb:6]: Can we add a test for this?

**Review by gpeng — CHANGES_REQUESTED** [review]

File: app/services/feature_service.rb (hunk 3 above)
Comment (darokel) [reply app/services/feature_service.rb:6]: done

**Review by defong — APPROVED** [review]
Minor thing...

Would the comments  littered here and there make more sense in the README?

File: app/services/feature_service.rb (hunk 3 above)
Comment (gpeng) [reply app/services/feature_service.rb:6]: Can we have a test for the nested feature splitting?

**Review by gpeng — CHANGES_REQUESTED** [review]

File: app/services/feature_service.rb (hunk 3 above)
Comment (darokel) [reply app/services/feature_service.rb:6]: done https://github.com/DFE-Digital/register-trainee-teacher-data/pull/44/commits/3ee1d0f6d02cca943c6fc1c97ee8f5513ba62319

**Comment by darokel** [comment]
> Minor thing...
> 
> Would the comments littered here and there make more sense in the README?

yup done in https://github.com/DFE-Digital/register-trainee-teacher-data/pull/44/commits/0f3cad596bf00f9991e228169f699c4ba9e4e050

**Review by gpeng — APPROVED** [review]

**Pull request merged**